Mathlib/Algebra/Module/Torsion.lean

```diff
@@ -591,6 +591,19 @@ instance (a : R) {S : Type*} [SMul S R] [SMul S M] [IsScalarTower S R M] [IsScal
     IsScalarTower S (R ⧸ R ∙ a) (torsionBy R M a) :=
   inferInstance
 
+/-- Given an `R`-module `M` and an element `a` in `R`, submodules of the `a`-torsion submodule of
+`M` do not depend on whether we take scalars to be `R` or `R ⧸ R ∙ a`. -/
+def submodule_torsionBy_orderIso (a : R) :
+    Submodule (R ⧸ R ∙ a) (torsionBy R M a) ≃o Submodule R (torsionBy R M a) :=
+  { restrictScalarsEmbedding R (R ⧸ R ∙ a) (torsionBy R M a) with
+    invFun := fun p ↦
+      { carrier := p
+        add_mem' := add_mem
+        zero_mem' := p.zero_mem
+        smul_mem' := by rintro ⟨b⟩; exact p.smul_mem b }
```

Comment:
This is surprising... I would've expected that you would need to explicitly use something about `torsionBy` on this line.

Reply:
This is the reverse direction (regarding an `R`-submodule as an `(R ⧸ R ∙ a)`-submodule) so this step is trivial.

Of course you're right that there must be something non-trivial somewhere!

IIUC, what's happening is that the forward direction uses `restrictScalarsEmbedding` which demands a `IsScalarTower R (R ⧸ R ∙ a) (torsionBy R M a)` instance. This is ultimately provided by the proof of [Module.IsTorsionBySet.isScalarTower](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Torsion.html#Module.IsTorsionBySet.isScalarTower) where indeed we see `Quotient.inductionOn'` invoked.